pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -342,6 +349,74 @@ public String getTaskType() {
     return pinotTaskConfigs;
   }
 
+  /**
+   * Segment selection result
+   */
+  private class SegmentsSelection {
+    private long _bucketMs;
+    private String _mergeLevel;
+    private boolean _hasSpilledOverData = false;
+    private boolean _hasUnmergedSegments = false;
+    // List of buckets of segments, segments in _selectedSegments[_firstUnmergedBuckets: ] will be picked to be merged
+    List<List<SegmentZKMetadata>> _selectedSegments = new ArrayList<>();
+    int _firstUnmergedBucket = 0;
+
+    SegmentsSelection(long bucketMs, String mergeLevel) {
+      _bucketMs = bucketMs;
+      _mergeLevel = mergeLevel;
+      _selectedSegments.add(new ArrayList<>());
+    }
+
+    /**
+     * Add segment to the selection result
+     *
+     * @return false if the selection result is sealed, otherwise return true
```

Comment:
I think that it's better to keep the regular contract for add (return true if successfully added, return false otherwise).  
For indicating `sealed result`, we can set the `_isResultSealed = true` and add the `isResultSealed()` function.

Reply:
Updated the comment that return `false` if the segment cannot be added, and this indicates that the selection result is sealed.